packages/synapse-react-client/src/components/GenericCard/TableRowGenericCard.tsx

```diff
@@ -0,0 +1,567 @@
+import { GenericCardIcon } from '@/components/GenericCard/GenericCardIcon'
+import { useGetEntity } from '@/synapse-queries'
+import { SynapseConstants } from '@/utils'
+import { calculateFriendlyFileSize } from '@/utils/functions/calculateFriendlyFileSize'
+import {
+  isDatasetCollection,
+  isTableEntity,
+} from '@/utils/functions/EntityTypeUtils'
+import { PRODUCTION_ENDPOINT_CONFIG } from '@/utils/functions/getEndpoint'
+import {
+  convertDoiToLink,
+  SYNAPSE_ENTITY_ID_REGEX,
+} from '@/utils/functions/RegularExpressions'
+import { GetAppTwoTone } from '@mui/icons-material'
+import { Collapse } from '@mui/material'
+import {
+  ColumnModel,
+  ColumnType,
+  ColumnTypeEnum,
+  Entity,
+  FileHandleAssociateType,
+  FileHandleAssociation,
+  SelectColumn,
+  Table,
+} from '@sage-bionetworks/synapse-types'
+import React, { useState } from 'react'
+import { ColumnIconConfigs, TargetEnum } from '../CardContainerLogic'
+import CitationPopover from '../CitationPopover'
+import { EntityDownloadConfirmation } from '../EntityDownloadConfirmation'
+import { HeaderCardVariant } from '../HeaderCard'
+import { IconOptions } from '../Icon'
+import IconList from '../IconList'
+import { useQueryContext } from '../QueryContext'
+import { useQueryVisualizationContext } from '../QueryVisualizationWrapper'
+import GenericCard from './GenericCard'
+import GenericCardActionButton from './GenericCardActionButton'
+import { SynapseCardLabel } from './SynapseCardLabel'
+
+export type TableToGenericCardMapping = {
+  /** The 'type' of resource a card refers to. Renders a label on the card with this string value */
+  type: string
+  /** The displayed title of the card */
+  title: string
+  /** The column name of the card title */
+  subTitle?: string
+  description?: string
+  includeCitation?: boolean
+  defaultCitationFormat?: 'bibtex' | 'apa' | 'ieee' | 'nature' | 'science'
+  citationBoilerplateText?: string
+  icon?: string
+  imageFileHandleColumnName?: string
+  thumbnailRequiresPadding?: boolean
+  secondaryLabels?: string[]
+  customSecondaryLabelConfig?: {
+    key: string
+    value: string
+    isVisible: (schema: Record<string, number>, data: string[]) => boolean
+  }
+
+  link?: string
+  dataTypeIconNames?: string
+  /** The column name whose data contains a synId that can be used to show a button to add the corresponding entity to the download cart. */
+  downloadCartSynId?: string
+}
+
+export type TableRowGenericCardProps = {
+  genericCardSchema: TableToGenericCardMapping
+  selectColumns?: SelectColumn[]
+  columnModels?: ColumnModel[]
+  iconOptions?: IconOptions
+  useTypeColumnForIcon?: boolean
+  isHeader?: boolean
+  headerCardVariant?: HeaderCardVariant
+  isAlignToLeftNav?: boolean
+  // Maps columnName to index
+  schema: Record<string, number>
+  // Row values
+  data: string[]
+  rowId?: number
+  columnIconOptions?: ColumnIconConfigs
+} & CommonCardProps
+
+export const getFileHandleAssociation = (
```

Comment:
no changes to utility function

Reply:
Does this makes sense to live in TableRowGenericCard, or to be moved to a utils ts file?  If it has the potential for reuse, I'd recommend refactoring.  Otherwise, it's fine.

Reply:
I think we've implemented this quite a few times, honestly. Definitely should be refactored